File: backend/app/api/schemas/audit.py
"""Contratos do modulo de auditoria."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.core.enums import AuditCategory, AuditResult


class AuditEventRead(BaseModel):
    id: uuid.UUID
    sequence: int
    occurred_at: datetime
    actor: str
    actor_role: str | None
    unit: str | None
    category: AuditCategory
    action: str
    resource_type: str
    resource_id: str | None
    result: AuditResult
    justification: str | None
    request_id: str | None
    analysis_id: str | None
    workflow_id: str | None
    job_id: str | None
    # Detalhe completo da acao (payload especifico de cada `action`, ex.:
    # modelo/provider/hash de entrada-saida de uma chamada de IA, filtros
    # de uma consulta, resultado por modalidade de um processamento) -
    # exibido na UI via popup dedicado (ver `AuditPage.tsx`), nunca editado
    # (a tabela e append-only). `event_hash`/`prev_hash` tambem expostos
    # aqui para permitir a mesma verificacao visual da cadeia de
    # integridade que `app.audit.hashing.verify_chain` faz
    # programaticamente.
    event_metadata: dict
    event_hash: str
    prev_hash: str | None

    model_config = {"from_attributes": True}

File: frontend/src/types/audit.ts
/** Espelha backend/app/api/schemas/audit.py. */
import type { AuditCategory, AuditResult } from "./enums.generated";

export interface AuditEvent {
  id: string;
  sequence: number;
  occurred_at: string;
  actor: string;
  actor_role: string | null;
  unit: string | null;
  category: AuditCategory;
  action: string;
  resource_type: string;
  resource_id: string | null;
  result: AuditResult;
  justification: string | null;
  request_id: string | null;
  analysis_id: string | null;
  workflow_id: string | null;
  job_id: string | null;
  /** Detalhe completo da acao (payload especifico de cada `action`),
   * exibido via popup dedicado - a tabela de auditoria e append-only,
   * nunca editada. */
  event_metadata: Record<string, unknown>;
  event_hash: string;
  prev_hash: string | null;
}
